src/KeyringController.ts

```diff
@@ -412,28 +486,46 @@ class KeyringController extends EventEmitter {
    * Attempts to sign the provided message parameters.
    * Prefixes the hash before signing per the personal sign expectation.
    *
-   * @param {object} msgParams - The message parameters to sign.
-   * @param {object} opts - Additional signing options.
-   * @returns {Promise<Buffer>} The raw signature.
+   * @param msgParams - The message parameters to sign.
+   * @param opts - Additional signing options.
+   * @returns The raw signature.
    */
-  async signPersonalMessage(msgParams, opts = {}) {
+  async signPersonalMessage(
+    msgParams: MessageParams,
+    opts: Record<string, unknown> = {},
+  ): Promise<string> {
     const address = normalizeAddress(msgParams.from);
     const keyring = await this.getKeyringForAccount(address);
-    return await keyring.signPersonalMessage(address, msgParams.data, opts);
+    if (!keyring.signPersonalMessage) {
+      throw new Error(KeyringControllerError.UnsupportedSignPersonalMessage);
+    }
+
+    return await keyring.signPersonalMessage(
+      address,
+      msgParams.data as Hex,
```

Comment:
Similar to other comments, is there any way we can avoid this typecast?

Reply:
[e36c134](https://github.com/MetaMask/KeyringController/pull/202/commits/e36c1348aadc882625deda681912d6b45786060e)